### PR TITLE
Rename `bevy_core::name::DebugName` to `bevy_core::name::NameOrEntity`

### LINKED ISSUE #14039: Rename bevy_core::name::DebugName to better fit its implementation
## What problem does this solve or what need does it fill?

The name of the `DebugName` struct is a bit of a misnomer: we don't have a separate struct for entity "debug names", and it just reuses the standard `Name` component, falling back to displaying the entity's ID if it doesn't have a name. We should rename `DebugName`.

## What solution would you like?

I suggest `NameOrId`: it remains short to spell out, and directly reflects its intentions. Welcome to other bikeshed names!

### PR DESCRIPTION
**Pull request by GauravTalreja**
# Objective

- Fixes #14039

## Solution

- Rename.

## Testing

- CI

---

## Migration Guide

- Rename usages of `bevy_core::name::DebugName` to `bevy_core::name::NameOrEntity`

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Comment by hymm** [comment]
This doesn't feel like an improvement. The name is now completely divorced from the fact that It queries for Option<&Name>. And you're less likely to discover it when you search for `Name` in the docs.

**Comment by GauravTalreja** [comment]
We're trying to convey that this struct is a query that optionally retrieves a `Name` or UID, often useful for identifying an entity for debug purposes.

I don't have a preference for which name is the more intuitive label, I leave it up to the maintainers.

**Comment by alice-i-cecile** [comment]
I think that `NameOrEntity` would be clearer: this isn't super tightly linked to debugging.

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**
